### PR TITLE
Undo Failed to attend not working for SJO screen (BE)

### PR DESCRIPTION
**Pull request by ricardofreitasrocha**
### Change description ###
The SJO screen should display undo failed to attend screen with the following search options:

* Juror Number
* Juror First name
* Juror last name
* Pool

The search shall fail if there is over 500 results ‘there are more than 500 results. Refine your search by adding more information'

A successful search will display the following columns for jurors that meet the criteria that have failed to attend status:

* Juror number (hyperlink)
* First name
* Last name
* Postcode

The system shall allow the SJO to select 1 or multiple jurors to undo the failed to attend 

The system shall display a check the SJO is sure they want to undo the failed to attend 'your undoing a failed to attend status for x jurors.

‘Their juror status will revert to ‘Responded’ and they must continue their jury service’

upon undoing the failed to attend the system shall display a success message ‘Remove failed to attend status for x juros’

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/SjoTasksControllerITest.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+        staff.setCourts(Collections.singletonList(TestConstants.VALID_COURT_LOCATION));
+
+        final String courtJwt = mintBureauJwt(BureauJwtPayload.builder()
+            .userType(UserType.COURT)
+            .login("COURT_USER")
+            .owner(TestConstants.VALID_COURT_LOCATION)
+            .locCode(TestConstants.VALID_COURT_LOCATION)
+            .roles(List.of(Role.MANAGER, Role.SENIOR_JUROR_OFFICER))
+            .staff(staff)
+            .build());
```
Comment (Ben-Edwards-cgi) [inline src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/SjoTasksControllerITest.java:80]: Should use the getCourtJwt(...) method

File: src/main/java/uk/gov/hmcts/juror/api/moj/controller/SjoTasksController.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+@RequiredArgsConstructor(onConstructor_ = {@Autowired})
+@IsSeniorCourtUser
+public class SjoTasksController {
+
+    private final SjoTasksService sjoTasksService;
+    private final JurorPoolService jurorPoolService;
+    private final BulkService bulkService;
+
+    @PostMapping("/juror/search")
+    @Operation(summary = "Get a list of failed to attend jurors based on search criteria")
```
Comment (Ben-Edwards-cgi) [inline src/main/java/uk/gov/hmcts/juror/api/moj/controller/SjoTasksController.java:39]: summary needs updating

File: src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/FailedToAttendListResponse.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Builder
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Schema(description = "Response containing juror details for failed to attend list")
+public class FailedToAttendListResponse {
```
Comment (Ben-Edwards-cgi) [inline src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/FailedToAttendListResponse.java:15]: Is this class needed?

File: src/main/java/uk/gov/hmcts/juror/api/moj/service/JurorHistoryService.java [hunk 4]
```diff
@@ -90,5 +88,9 @@ void createReassignPoolMemberHistory(JurorPool sourceJurorPool, String targetPoo
 
     void createPoolEditHistory(JurorPool updatedPool);
 
+    public String getHistoryDescription(String historyCode);
```
Comment (Ben-Edwards-cgi) [inline src/main/java/uk/gov/hmcts/juror/api/moj/service/JurorHistoryService.java:91]: public not needed

File: src/test/java/uk/gov/hmcts/juror/api/moj/controller/CompleteServiceControllerTest.java [hunk 5]
```diff
@@ -81,6 +79,8 @@ class CompleteServiceControllerTest {
     @MockBean
     private CompleteServiceService completeServiceService;
 
+    @MockBean
+    private JurorPoolService jurorPoolService;
```
Comment (Ben-Edwards-cgi) [inline src/test/java/uk/gov/hmcts/juror/api/moj/controller/CompleteServiceControllerTest.java:83]: Is this needed?

File: src/test/java/uk/gov/hmcts/juror/api/moj/service/JurorRecordServiceTest.java [hunk 6]
```diff
@@ -220,6 +220,8 @@ class JurorRecordServiceTest {
     private UserServiceModImpl userServiceMod;
     @Mock
     private JurorPaymentsSummaryRepository jurorPaymentsSummaryRepository;
+    @Mock
+    private SjoTasksService sjoTasksService;
```
Comment (Ben-Edwards-cgi) [inline src/test/java/uk/gov/hmcts/juror/api/moj/service/JurorRecordServiceTest.java:224]: Is this needed?

**Review by Ben-Edwards-cgi — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/juror/api/moj/service/JurorRecordServiceTest.java (hunk 6 above)
Comment (ricardofreitasrocha) [reply src/test/java/uk/gov/hmcts/juror/api/moj/service/JurorRecordServiceTest.java]: updated

File: src/test/java/uk/gov/hmcts/juror/api/moj/controller/CompleteServiceControllerTest.java (hunk 5 above)
Comment (ricardofreitasrocha) [reply src/test/java/uk/gov/hmcts/juror/api/moj/controller/CompleteServiceControllerTest.java]: not really 😄 removed

File: src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/FailedToAttendListResponse.java (hunk 3 above)
Comment (ricardofreitasrocha) [reply src/main/java/uk/gov/hmcts/juror/api/moj/controller/response/FailedToAttendListResponse.java]: nope. you're spot on, we're using the other common dto (JurorDetailsDto) 😄

File: src/main/java/uk/gov/hmcts/juror/api/moj/controller/SjoTasksController.java (hunk 2 above)
Comment (ricardofreitasrocha) [reply src/main/java/uk/gov/hmcts/juror/api/moj/controller/SjoTasksController.java]: updated

File: src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/SjoTasksControllerITest.java (hunk 1 above)
Comment (ricardofreitasrocha) [reply src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/SjoTasksControllerITest.java]: updated

File: src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/CompleteServiceControllerITest.java [hunk 7]
```diff
@@ -784,18 +784,18 @@ void negativeUnauthorisedNotSjo() throws Exception {
     @Sql({"/db/mod/truncate.sql", "/db/CompleteServiceControllerSearch.sql"})
     @SuppressWarnings("PMD.TooManyMethods")
     class GetCompleteJurors {
-        public static final String URL = BASE_URL;
+        public static final String URL = "/api/v1/moj/sjo-tasks/juror/search";
```
[the remark this answers is not in the document]
Comment (ricardofreitasrocha) [reply src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/CompleteServiceControllerITest.java]: moved tests to route parent class

**Review by Ben-Edwards-cgi — APPROVED** [review]

**Pull request merged**